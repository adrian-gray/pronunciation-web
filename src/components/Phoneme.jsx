import React from 'react'

const Phoneme = (props) => (
  <div>
    <p>I am a phoneme</p>
  </div>
)

export default Phoneme
