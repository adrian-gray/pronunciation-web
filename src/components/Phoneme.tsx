import React from "react";
import { Link } from "react-router-dom";

import Container from "react-bootstrap/Container";

import SplitHilite from "./SplitHilite";

export default (props: {
  phoneme: string;
  ipa: string;
  tag: string;
  title: string;
  words: string[];
}) => {
  const { phoneme, ipa, tag, title, words } = props;
  const displayTitle = (
    <span>
      <SplitHilite str={ipa} /> – <SplitHilite str={title} />
    </span>
  );
  const wordList = words
    .slice(0, 6)
    .map((word: string) => <SplitHilite str={word} key={word} />);

  return (
    <Container>
      <Link to={{ pathname: `/sound/${phoneme}` }} className="undecorated">
        <div>
          <h3>{displayTitle}</h3>
          <p>{"Example words containing short ‘a’"}</p>
          <p className="larger-text">{wordList}</p>
          <h5>{`Click to learn more about the ${tag} sound`}</h5>
        </div>
      </Link>
    </Container>
  );
};
