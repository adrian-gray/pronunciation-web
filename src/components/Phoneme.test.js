import React from "react";
import { shallow } from "enzyme";
import { render, cleanup } from "@testing-library/react";

import Phoneme from "./Phoneme";

afterEach(cleanup);

test("Phoneme", async () => {
  const props = {};

  const container = shallow(<Phoneme {...props} />);

  expect(container.firstChild).toMatchSnapshot();
});
