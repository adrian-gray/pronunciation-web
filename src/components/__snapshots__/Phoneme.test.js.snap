// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phoneme 1`] = `
<div>
  <div
    class="headspace"
  >
    <a
      class="undecorated"
      href="/sound/long-z"
    >
      <div
        class="mini-canvas"
      >
        <h3>
          <span>
            <span>
              z:!
            </span>
             
             – 
            <span>
              Long Zorro
            </span>
             
          </span>
        </h3>
        <p>
          Example words containing short ‘a’
        </p>
        <p
          class="larger-text"
        >
          <span>
            zoo
          </span>
           
          <span>
            zork
          </span>
           
          <span>
            zorb
          </span>
           
        </p>
        <h5>
          Click to learn more about the zzzz sound
        </h5>
      </div>
    </a>
  </div>
</div>
`;
